Controller/Admin/TransactionTabPostProcessing.php

```diff
@@ -150,56 +183,59 @@ private function _processRequest(array $aRequestParameters)
      */
     protected function _getPostProcessingActions(): array
     {
-        // TODO return only supported actions for the payment
-        return [
-            'cancel' => [
-                'title' => Helper::translate('cancel'),
-                'callback' => '_onCancel',
-            ],
-            'capture' => [
-                'title' => Helper::translate('pay'),
-                'callback' => '_onCapture',
-            ],
-            'refund' => [
-                'title' => Helper::translate('refund'),
-                'callback' => '_onRefund',
-            ],
-        ];
-    }
+        $oPaymentMethod = Payment_Method_Factory::create($this->oTransaction->getPaymentType());
 
-    /**
-     * Callback function for the 'cancel' action.
-     *
-     * @param float $fAmount
-     * @return string
-     */
-    private function _onCancel(float $fAmount): string
-    {
-        // TODO
-        return Helper::translate('text_generic_success');
+        $oTransaction = $oPaymentMethod->getTransaction();
+
+        $sParentTransactionId = $this->oTransaction->wdoxidee_ordertransactions__transactionid->value;
+        $oTransaction->setParentTransactionId($sParentTransactionId);
+
+        // load the supported operations array from the backend service
+        $oBackendService = new BackendService($oPaymentMethod->getConfig(), $this->_oLogger);
+        $aSupportedOperations = $oBackendService->retrieveBackendOperations($oTransaction, true);
+
+        $aOperations = array();
+
+        foreach ($aSupportedOperations as $key => $value) {
+            $aOperation = array(
+                'action' => $key,
+                'title' => Helper::translate($value),
```

Comment:
Can we add an array above, used as a look-up table? e.g.:

```
$aOperationTitles = array(
    'refund' => Helper::translate('refund'),
    // more actions here...
);
```
Then, in the loop:
```
'title' => $aOperationTitles[$value],
```

The reason I'm suggesting this is that we rely on `Helper::translate()` to contain **keys** as arguments, when we parse the code for the PhraseApp integration.

Reply:
👍